azure-pipelines.yaml

```diff
@@ -5,10 +5,24 @@ trigger:
 
 jobs:
   - job: Release
-    condition: and(eq(variables['Build.Reason'], 'Manual'), eq(variables['release'], 'true')))
+    condition: or(eq(variables['release'], 'true'), ne(variables['Build.SourceBranchName'], 'master'))
     pool:
       vmImage: 'ubuntu-16.04'
     variables:
     - group: GitHub
     steps:
-    - bash: echo "Hello World"
+    - bash: |
+        # By default a release build on master won't do a dry-run except you explictly told so.
```

Comment:
Why not just always run dry-run if the job is triggered *unless* release is set?